### PR TITLE
remove `test-all` and add `make-watch` rule

### PR DESCRIPTION
**Pull request by retrofox**
In this PR I've changed slightly the tests rules:

### Run all tests

```bash
> make test
```

### Watch, filter and run tests

```bash
> FILTER=wpcom.site.settings make test-watch
```

**screenshot**
image

**video**
https://cloudup.com/csmrGGDNHpB

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -46,20 +45,21 @@ test:
...
+test-watch:
 	@$(MOCHA) \
 		--timeout 120s \
 		--slow 3s \
 		--bail \
 		--reporter spec \
 		--compilers js:babel-register \
-		test
+		--watch \
+		--grep "$(filter-out $@,$(MAKECMDGOALS))" \
```
Comment (TooTallNate) [inline Makefile]: This is not a good pattern, which was why I introduced the `FILTER` variable previously. Please see http://stackoverflow.com/a/6273809.

File: package.json [hunk 2]
```diff
@@ -20,6 +20,7 @@
     "browser",
     "rest",
     "api",
+    "request handler",
```
Comment (TooTallNate) [inline package.json]: This seems kind of random…

**Comment by retrofox** [comment]
@TooTallNate updated and rebased

File: Makefile [hunk 3]
```diff
@@ -46,20 +45,21 @@ test:
 	@$(MOCHA) \
 		--timeout 120s \
 		--slow 3s \
-		--grep "$(FILTER)" \
```
Comment (TooTallNate) [inline Makefile]: Why removing this one?

File: test/test.apiVersion.js [hunk 4]
```diff
@@ -1,33 +1,33 @@
...
+ * Internal dependencies
+ */
+import util from './util';
 
 /**
  * Fixture data
  */
-
-var fixture = require( './fixture' );
+const fixture = require( './fixture' );
```
Comment (TooTallNate) [inline test/test.apiVersion.js]: `import`?

File: Makefile (hunk 1 above)
Comment (retrofox) [inline Makefile]: Thanks Nate.

File: Makefile (hunk 3 above)
Comment (retrofox) [inline Makefile]: In theory `make tests` is used to run all tests. And if you wanna filter some tests you would like to use test-watch. At least I thought that. Anyway ... no make sense remove --grep here so I've restored this parameter.

File: test/test.apiVersion.js (hunk 4 above)
Comment (retrofox) [inline test/test.apiVersion.js]: I didn't know that you could import directly a json file.

**Comment by TooTallNate** [comment]
LGTM 👯

**Comment by retrofox** [comment]
thanks! nice emoji btw

**Pull request merged**